src/rgw/rgw_bucket_layout.h

```diff
@@ -87,6 +87,20 @@ inline std::ostream& operator<<(std::ostream& out, const BucketLogType &log_type
   }
 }
 
+inline auto matches_gen(uint64_t gen)
+{
+  return [gen] (const bucket_log_layout_generation& l) { return l.gen == gen; }
+}
+
+inline bucket_index_layout_generation log_to_index_layout(const bucket_log_layout_generation& log_layout)
+{
+  ceph_assert(log_layout.layout.type == BucketLogType::InIndex);
+  bucket_index_layout_generation index;
+  index.gen = log_layout.layout.in_index.gen;
+  index.layout.normal = log_layout.layout.in_index.layout;
+  return index;
+}
+
```

Comment:
these functions need to move below the declaration of `struct bucket_log_layout_generation`